### ISSUE TITLE
[hbo] Improve extractor support for different URL scheme

### ISSUE THREAD

**Issue by SeanPM5** [rg3/youtube-dl#10892]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.10.07**
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Site support enhancement

### Description of your *issue*, suggested solution and other information
I noticed there is already an extractor for some HBO videos, but it only works when the URL is in a specific format ending with a video ID.  There are many videos on HBO website that do not contain a video ID in their URL (examples listed below).  However if you view page source on these and search for "videoId" you will find the video ID for it, which you can then use to re-construct a new working URL according to extractor requirements.

So basically if you can add these types of links to valid URL list on HBO Extractor.

### Example URLs:
- Single video: http://www.hbo.com/westworld/about/video/westworld-teaser-trailer.html?autoplay=true
- Single video: http://www.hbo.com/girls/episodes/5/52-i-love-you-baby/video/ep-52-inside-the-episode.html?autoplay=true
- Single video: http://www.hbo.com/game-of-thrones/about/video/season-5-invitation-to-the-set.html?autoplay=true

**Comment by yan12125** [comment]
Thanks for the report. Those URLs will be supported in the next version.

**Comment by SeanPM5** [comment]
Great, thanks very much for adding this!
